src/libraries/Common/tests/System/Security/Cryptography/MLKemTests.cs

```diff
@@ -164,6 +164,56 @@ public static void ImportSubjectPublicKeyInfo_NullSource()
                 MLKem.ImportSubjectPublicKeyInfo((byte[])null));
         }
 
+        [Fact]
+        public static void ImportSubjectPublicKeyInfo_WrongAlgorithm()
+        {
+            byte[] ecP256Spki = Convert.FromBase64String(@"
+                MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEuiPJ2IV089LVrXZGDo9Mc542UZZE
+                UtPQVd60Ckb/u5OXHAlmITVzFPThKI+N/bUMEnnHEmF8ZDUtLiQPBaKiMQ==");
+            Assert.Throws<CryptographicException>(() => MLKem.ImportSubjectPublicKeyInfo(ecP256Spki));
+        }
+
+        [Fact]
+        public static void ImportSubjectPublicKeyInfo_WrongParameters()
+        {
+            byte[] mlKem512BadParameters = (
```

Comment:
Do we have similar tests for all three Pkcs8 content-formats (and then the encrypted ones atop them)?

Reply:
They exist for PKCS#8, e.g.

* https://github.com/dotnet/runtime/blob/15d227ca48b689bf731b22d0346823b10adfd9a7/src/libraries/Common/tests/System/Security/Cryptography/MLKemTests.cs#L234
* https://github.com/dotnet/runtime/blob/15d227ca48b689bf731b22d0346823b10adfd9a7/src/libraries/Common/tests/System/Security/Cryptography/MLKemTests.cs#L258-L263

I didn't do them specifically for Encrypted PKCS#8 since the (already tested) PKCS#8 importers are their workhorse. I can add them if you would like.

Reply:
If you want to add it for black box virtue, go for it; but I'm content to whitebox that.